File: TWS_API/samples/CSharp/Testbed/NewsSamplesProto.cs
/* Copyright (C) 2026 Interactive Brokers LLC. All rights reserved. This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using IBApi;

namespace Samples
{
    public class NewsSamplesProto
    {
        public static IBApi.protobuf.HistoricalNewsRequest HistoricalNewsRequestWithEndTime(int reqId)
        {
            //! [historical_news_request_with_end_time]
            IBApi.protobuf.HistoricalNewsRequest historicalNewsRequestProto = new IBApi.protobuf.HistoricalNewsRequest();
            historicalNewsRequestProto.ReqId = reqId;
            historicalNewsRequestProto.ConId = 8314;
            historicalNewsRequestProto.ProviderCodes = "BRFUPDN+BRFG";
            DateTime endDateTime = DateTime.Now.AddDays(-10);
            historicalNewsRequestProto.EndDateTime = endDateTime.ToString("yyyy-MM-dd HH:mm:ss.0");
            historicalNewsRequestProto.TotalResults = 10;
            return historicalNewsRequestProto;
            //! [historical_news_request_with_end_time]
        }

        public static IBApi.protobuf.HistoricalNewsRequest HistoricalNewsRequestWithStartTime(int reqId)
        {
            //! [historical_news_request_with_start_time]
            IBApi.protobuf.HistoricalNewsRequest historicalNewsRequestProto = new IBApi.protobuf.HistoricalNewsRequest();
            historicalNewsRequestProto.ReqId = reqId;
            historicalNewsRequestProto.ConId = 8314;
            historicalNewsRequestProto.ProviderCodes = "BRFUPDN+BRFG";
            DateTime startDateTime = DateTime.Now.AddDays(-10);
            historicalNewsRequestProto.StartDateTime = startDateTime.ToString("yyyy-MM-dd HH:mm:ss.0");
            historicalNewsRequestProto.TotalResults = 10;
            return historicalNewsRequestProto;
            //! [historical_news_request_with_start_time]
        }
    }
}

File: TWS_API/samples/VB/Testbed/NewsSamplesProto.vb
' Copyright (C) 2026 Interactive Brokers LLC. All rights reserved. This code is subject to the terms
' and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable.


Namespace Samples

    Public Class NewsSamplesProto

        Public Shared Function HistoricalNewsRequestWithEndTime(reqId As Integer) As IBApi.protobuf.HistoricalNewsRequest
            '! [historical_news_request_with_end_time]
            Dim historicalNewsRequestProto As New IBApi.protobuf.HistoricalNewsRequest()
            historicalNewsRequestProto.ReqId = reqId
            historicalNewsRequestProto.ConId = 8314
            historicalNewsRequestProto.ProviderCodes = "BRFUPDN+BRFG"
            historicalNewsRequestProto.EndDateTime = String.Format("{0:yyyy-MM-dd HH:mm:ss.0}", DateTime.Now.AddDays(-10))
            historicalNewsRequestProto.TotalResults = 10
            Return historicalNewsRequestProto
            '! [historical_news_request_with_end_time]
        End Function

        Public Shared Function HistoricalNewsRequestWithStartTime(reqId As Integer) As IBApi.protobuf.HistoricalNewsRequest
            '! [historical_news_request_with_start_time]
            Dim historicalNewsRequestProto As New IBApi.protobuf.HistoricalNewsRequest()
            historicalNewsRequestProto.ReqId = reqId
            historicalNewsRequestProto.ConId = 8314
            historicalNewsRequestProto.ProviderCodes = "BRFUPDN+BRFG"
            historicalNewsRequestProto.StartDateTime = String.Format("{0:yyyy-MM-dd HH:mm:ss.0}", DateTime.Now.AddDays(-10))
            historicalNewsRequestProto.TotalResults = 10
            Return historicalNewsRequestProto
            '! [historical_news_request_with_start_time]
        End Function

    End Class

End Namespace
